src/main/org/openjfx/controller/MainController.java

```diff
@@ -10,98 +10,109 @@
 import javafx.scene.Node;
 import javafx.scene.control.Button;
 import javafx.scene.control.ChoiceBox;
+import javafx.scene.control.TextArea;
 import javafx.scene.control.TextField;
 import javafx.stage.Stage;
 import org.openjfx.model.SerialPortService;
 import org.openjfx.model.settings.PortSettings;
 import org.openjfx.model.SceneLoader;
 
 import java.net.URL;
+import java.util.List;
 import java.util.ResourceBundle;
 
 public class MainController implements Initializable {
 
     @FXML
     public ChoiceBox<Integer> baudRate;
-
     @FXML
     public ChoiceBox<Integer> dataFieldLength;
-
     @FXML
     public ChoiceBox<Integer> stop;
-
     @FXML
     public TextField terminatorChar;
-
     @FXML
     public ChoiceBox<String> port;
-
     @FXML
     public ChoiceBox<Character> parity;
-
     @FXML
     public ChoiceBox<String> terminator;
-
-    @FXML
-    public Button btn_receive;
-
     @FXML
     public ChoiceBox<String> flowControl;
+    @FXML
+    public Button sendBtn;
+    @FXML
+    public TextArea receivedText;
+    @FXML
+    public Button openPortBtn;
+    @FXML
+    public Button closePortBtn;
+    @FXML
+    public Button pingBtn;
+    @FXML
+    public TextArea inputText;
+
+    private SerialPortService portService = new SerialPortService();
 
     @FXML
-    public Button btn_send;
+    public void ping(ActionEvent event) {
+        receivedText.setText(portService.ping(PortSettings.openedPort));
+    }
 
+    @FXML
+     public void sendText(ActionEvent event) {
+        String messageToSend = inputText.getText() + appendTerminatorCharacters(PortSettings.getTerminatorChars());
+        portService.sendString(PortSettings.openedPort, messageToSend);
+    }
 
+    private String appendTerminatorCharacters(List<Character> terminatorCharacters) {
+        StringBuilder result = new StringBuilder();
+        for(Character character : terminatorCharacters) {
+            result.append(character);
+        }
+        return result.toString();
+    }
 
     @FXML
-    public void receive_chosen(ActionEvent event) throws Exception {
+    public void openPort(ActionEvent event) {
         fillSettings();
-        Stage stage = (Stage)((Node)event.getSource()).getScene().getWindow();
-        SceneLoader.loadScene("view/receive_mode.fxml", "receive", stage);
+       // PortSettings.openedPort = new SerialPortService().getInitializedPort(new PortSettings(),this);
```

Comment:
to ma być zakomentowane?

Reply:
Tak, bo narazie nie ma listenera więc nie ma czego tam przekazać jako drugi argument